From 90d1e7ee5ac72ccec1f03ca717f2185feeea31d9 Mon Sep 17 00:00:00 2001
From: Paul Keith <javelinanddart@gmail.com>
Date: Tue, 10 Apr 2018 21:35:32 +0200
Subject: [PATCH 1/2] klte-common: Use passthrough manifest for all NFC chips

* And make a common nfc board stub while we're at it

Change-Id: I4ef9c69ebbff539924ba4dc10c8035521dbf5424
---
 nfc/bcm2079x/board.mk                          |  2 +-
 nfc/board.mk                                   | 18 ++++++++++++++++++
 nfc/manifest-hwbinder.xml                      | 11 -----------
 nfc/{manifest-passthrough.xml => manifest.xml} |  0
 nfc/pn547/board.mk                             |  5 +++--
 nfc/s3fwrn5/board.mk                           |  2 +-
 6 files changed, 23 insertions(+), 15 deletions(-)
 create mode 100644 nfc/board.mk
 delete mode 100644 nfc/manifest-hwbinder.xml
 rename nfc/{manifest-passthrough.xml => manifest.xml} (100%)

diff --git a/nfc/bcm2079x/board.mk b/nfc/bcm2079x/board.mk
index 8b83084..4c8a742 100644
--- a/nfc/bcm2079x/board.mk
+++ b/nfc/bcm2079x/board.mk
@@ -15,4 +15,4 @@
 #
 
 # NFC
-DEVICE_MANIFEST_FILE += $(COMMON_PATH)/nfc/manifest-hwbinder.xml
+include $(COMMON_PATH)/nfc/board.mk
diff --git a/nfc/board.mk b/nfc/board.mk
new file mode 100644
index 0000000..8f7f475
--- /dev/null
+++ b/nfc/board.mk
@@ -0,0 +1,18 @@
+#
+# Copyright (C) 2018 The LineageOS Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# NFC
+DEVICE_MANIFEST_FILE += $(COMMON_PATH)/nfc/manifest.xml
diff --git a/nfc/manifest-hwbinder.xml b/nfc/manifest-hwbinder.xml
deleted file mode 100644
index 4c4d7a1..0000000
--- a/nfc/manifest-hwbinder.xml
+++ /dev/null
@@ -1,11 +0,0 @@
-<manifest version="1.0" type="device">
-    <hal format="hidl">
-        <name>android.hardware.nfc</name>
-        <transport>hwbinder</transport>
-        <version>1.0</version>
-        <interface>
-            <name>INfc</name>
-            <instance>default</instance>
-        </interface>
-    </hal>
-</manifest>
diff --git a/nfc/manifest-passthrough.xml b/nfc/manifest.xml
similarity index 100%
rename from nfc/manifest-passthrough.xml
rename to nfc/manifest.xml
diff --git a/nfc/pn547/board.mk b/nfc/pn547/board.mk
index 046d76a..e52b1ab 100644
--- a/nfc/pn547/board.mk
+++ b/nfc/pn547/board.mk
@@ -15,7 +15,8 @@
 #
 
 # NFC
+include $(COMMON_PATH)/nfc/board.mk
+
+# PN547 specific flags
 BOARD_NFC_CHIPSET := pn547
 BOARD_NFC_HAL_SUFFIX := msm8974
-
-DEVICE_MANIFEST_FILE += $(COMMON_PATH)/nfc/manifest-passthrough.xml
diff --git a/nfc/s3fwrn5/board.mk b/nfc/s3fwrn5/board.mk
index a7a7378..4c8a742 100644
--- a/nfc/s3fwrn5/board.mk
+++ b/nfc/s3fwrn5/board.mk
@@ -15,4 +15,4 @@
 #
 
 # NFC
-DEVICE_MANIFEST_FILE += $(COMMON_PATH)/nfc/manifest-passthrough.xml
+include $(COMMON_PATH)/nfc/board.mk
-- 
2.17.0

